src/main/java/com/sonyericsson/jenkins/plugins/bfa/db/MongoDBKnowledgeBase.java

```diff
@@ -1022,19 +1022,22 @@ private void addIndicationsToDBObject(DBObject object, List<FoundIndication> ind
      */
     private MongoClient getMongoConnection() {
         if (mongo == null) {
+            MongoClientOptions mongoClientOptions = MongoClientOptions
+                    .builder()
+                    .connectTimeout(5000)
+                    .serverSelectionTimeout(5000)
```

Comment:
default timeout is 30 seconds, test connection isn't very nice with that. 5 seconds is a much better balance imo.

Reply:
Sounds OK. We might want the timeout configurable in the future, but that shouldn't be hard to add.